modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java

```diff
@@ -160,8 +160,14 @@ public void processOpts() {
             LOGGER.debug("Serialization library not set, using default {}", SERIALIZATION_LIBRARY_DEFAULT);
         }
         setLibrary(additionalProperties.get(CodegenConstants.SERIALIZATION_LIBRARY).toString());
+        
         if (SERIALIZATION_LIBRARY_BUILT_VALUE.equals(library)) {
-            this.setLegacyDiscriminatorBehavior(false);
```

Comment:
I did a search and found that other generators also set `setLegacyDiscriminatorBehavior` to false.

My guess is that this is introduced for backward compatibility so I suggest we keep it this way (set to false) if no user has presented a use case that may require setting setLegacyDiscriminatorBehavior to true.

Reply:
alright then I will close this PR for now, since I introduced another PR to change the default behavior